src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -26,29 +35,186 @@ const ActorInit Item_Etcetera_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85A98.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85B28.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85B6C.s")
+static s16 sObjectIds[] = {
+    OBJECT_GI_BOTTLE, OBJECT_GI_BOTTLE_LETTER, OBJECT_GI_SHIELD_2, OBJECT_GI_ARROWCASE, OBJECT_GI_SCALE,
+    OBJECT_GI_SCALE,  OBJECT_GI_KEY,           OBJECT_GI_M_ARROW,  OBJECT_GI_RUPY,      OBJECT_GI_RUPY,
+    OBJECT_GI_RUPY,   OBJECT_GI_RUPY,          OBJECT_GI_HEARTS,   OBJECT_GI_KEY,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/ItemEtcetera_Update.s")
+// Indexes passed to the item table in z_draw.c
+static s16 sDrawItemTableIdx[] = {
+    0x0000, 0x0044, 0x002B, 0x0015, 0x0029, 0x002A, 0x0001, 0x005F, 0x006C, 0x006D, 0x006E, 0x0070, 0x0013, 0x0001,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85C64.s")
+static s16 sGetItemId[] = {
+    GI_BOTTLE,     GI_LETTER_RUTO, GI_SHIELD_HYLIAN, GI_QUIVER_40, GI_SCALE_SILVER, GI_SCALE_GOLD, GI_KEY_SMALL,
+    GI_ARROW_FIRE, GI_INVALID,     GI_INVALID,       GI_INVALID,   GI_INVALID,      GI_INVALID,    GI_INVALID,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85CB8.s")
+void ItemEtcetera_SetupAction(ItemEtcetera* this, ItemEtceteraActionFunc actionFunc) {
+    this->actionFunc_15C = actionFunc;
+}
+
+void ItemEtcetera_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemEtcetera* this = THIS;
+    s32 pad;
+    s32 type;
+    s32 objBankIndex;
+
+    type = this->actor.params & 0xFF;
+    osSyncPrintf("no = %d\n", type);
+    objBankIndex = Object_GetIndex(&globalCtx->objectCtx, sObjectIds[type]);
+    osSyncPrintf("bank_ID = %d\n", objBankIndex);
+    if (objBankIndex < 0) {
+        __assert("0", "../z_item_etcetera.c", 241);
+    } else {
+        this->objBankIndex = objBankIndex;
+    }
+    this->drawId = sDrawItemTableIdx[type];
+    this->getItemId = sGetItemId[type];
+    this->actionFunc_14C = func_80B85824;
+    this->drawFunc = ItemEtcetera_DrawFireArrow;
+    Actor_SetScale(&this->actor, 0.25f);
+    ItemEtcetera_SetupAction(this, func_80B857D0);
+    switch (type) {
+        case ITEM_ETCETERA_LETTER:
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actionFunc_14C = func_80B858B4;
+            if (gSaveContext.eventChkInf[3] & 2) {
+                Actor_Kill(&this->actor);
+            }
+            break;
+        case ITEM_ETCETERA_ARROW_FIRE:
+            this->actionFunc_14C = ItemEtcetera_UpdateFireArrow;
+            Actor_SetScale(&this->actor, 0.5f);
+            this->actor.draw = NULL;
+            this->actor.shape.unk_08 = 50.0f;
+            break;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+        case 12:
+        case 13: // unused values?
```

Comment:
I believe these types are the ones used in the treasure chest minigame.
Looking at the draw function, you can see they are only drawn when the lens is active (actorCtx.unk_03 being set). And the object IDs + draw indexes correspond to (in order): Green Rupee, Blue Rupee, Red Rupee, Purple Rupee, Piece of Heart and Small Key.
I suggest naming types for these in this format: `ITEM_ETCETERA_RUPEE_GREEN_CMG`, `ITEM_ETCETERA_RUPEE_BLUE_CMG`, etc. CMG being short for chest minigame.

Reply:
Id go with the latter even though its long, just cause its not a common acronym that anyone will know what it means. Even with context of what the actor is doing imo itll be confusing

Reply:
CMG is fairly common or at least it's the most common one for this minigame afaik.

Reply:
maybe in the rando scene? coming from speedrunning this is never abbreviated at all 
but ye either way I think itll be fine without

Reply:
Yeah mostly the rando scene. 
35+ characters for a name is a bit much but I guess it's not a huge issue.

Reply:
will probably look nicer when we arent calling this thing ItemEtcetera 
but then again who knows what the readable name will be lol